src/App.tsx

```diff
@@ -1,14 +1,60 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { TodoModal } from './components/TodoModal';
+import { Todo } from './types/Todo';
+import { FilteredOptions } from './types/FilteredOption';
+import { getTodos } from './api';
+
+function handleFilteredTodos(
+  todos: Todo[],
+  selectedOption: FilteredOptions,
+  query: string,
+) {
+  let filteredTodos = [...todos];
+
+  if (query) {
+    filteredTodos = filteredTodos.filter(todo =>
+      todo.title.toLowerCase().includes(query.toLowerCase().trim()),
+    );
+  }
+
+  switch (selectedOption) {
+    case FilteredOptions.active:
+      return filteredTodos.filter(todo => !todo.completed);
+    case FilteredOptions.completed:
+      return filteredTodos.filter(todo => todo.completed);
+
+    default:
+      return filteredTodos;
+  }
```

Comment:
1. No need to copy anything
2. Please try implement it with a single `filter` call


```suggestion
 return filteredTodos.filter(todo => {
 
 switch (selectedOption) {
    case FilteredOptions.active:
      return !todo.completed'
    case FilteredOptions.completed:
      return todo.completed;

    default:
      return true;
  }
 
 // TODO: implement check on query
 
 })
```